compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -217,6 +217,8 @@
     private SymbolEnv env;
     private final boolean projectAPIInitiatedCompilation;
 
+    private boolean isField = false;
```

Comment:
Shall we add a `flag` in the NodeTransformer indicating that this is a `field` and ditch this?